KsApi/Service+RequestHelpers.swift

```diff
@@ -24,6 +25,34 @@ extension Service {
     .flatMap(self.decodeModel)
   }
 
+  func request<M: Decodable>(_ route: Route) -> AnyPublisher<M, ErrorEnvelope> {
+    let properties = route.requestProperties
+
+    guard let URL = URL(string: properties.path, relativeTo: self.serverConfig.apiBaseUrl as URL) else {
+      fatalError(
+        "URL(string: \(properties.path), relativeToURL: \(self.serverConfig.apiBaseUrl)) == nil"
+      )
+    }
+
+    return Service.session.combine_dataResponse(
+      preparedRequest(forURL: URL, method: properties.method, query: properties.query),
+      uploading: properties.file.map { ($1, $0.rawValue) },
+      and: self.perimeterXClient
+    ).tryMap { data in
+      let result: Result<M?, ErrorEnvelope> = self.decodeModel(data: data, ofType: M.self)
+
+      switch result {
+      case let .success(value):
+        return value
+      case let .failure(error):
+        throw error
+      }
+    }.compactMap { $0 }
+      .mapError { error in
+        error as! ErrorEnvelope
```

Comment:
`self.decodeModel` _only_ outputs `ErrorEnvelope` so this forced cast should be safe.